hyperspy/axes.py

```diff
@@ -114,11 +114,39 @@ def _convert_compact_units(self, factor=0.25, inplace=True):
     _convert_compact_units.__doc__ %= FACTOR_DOCSTRING
 
     def _get_index_from_value_with_units(self, value):
+        "Get axes index by passing a calibrated value with a unit, like '20nm'"
         value = _ureg.parse_expression(value)
         if not hasattr(value, 'units'):
             raise ValueError('"{}" should contains an units.'.format(value))
         return self.value2index(value.to(self.units).magnitude)
 
+    def _get_index_from_relative_string(self, relative_string):
+        "Get axes index by relative indexing using string like 'rel0.5'"
+        try:
+            assert relative_string[:3] == 'rel'
+            relative_value = float(relative_string[3:])
+        except ValueError:
+           print(f"Expected relative units of kind 'rel0.2' but '{relative_string}' could not be interpreted this way.")
+           raise
+        return self.get_index_from_relative_value(relative_value)
+
+    def get_value_from_relative_value(self, relative_value, nearest_axis_value=False):
+        '''
+        Gets the value of a position along the axis by relative or fractional value. The value is 
+        not necessarily in the .axis array.
+
+        If nearest_axis_value is True, the value returned is the nearest value that is on .axis
+        '''
+        value = self.low_value + relative_value * (self.high_value - self.low_value)
+        if nearest_axis_value:
+            value = self.index2value(self.value2index(value))
+        return value
+
+    def get_index_from_relative_value(self, relative_value):
+        "Get the index of a position in the axis by relative or fractional value"
+        value = self.get_value_from_relative_value(relative_value)
+        return self.value2index(value)
```

Comment:
IInstead to adding these two methods, wouldn't it more simple to allow `value2index` to take "string" value?

Reply:
Yes, and then for the first method, one would do `index2value(value2index(value)))`.
I'll take a look at doing the same for values with units.

Reply:
Yes, or alternatively, use the `axis` attribute of `DataAxis`: `ax.axis[ax.value2index(value)]`
But what would be the use case for the first method?

Reply:
The usecase was just general convenience, for instance for getting the central half (0.25 to 0.75) of an axes for plotting something with matplotlib on top of a signal.